packages/langium-cli/src/generate.ts

```diff
@@ -365,10 +365,21 @@ export async function runGenerator(config: LangiumConfig, options: GenerateOptio
         }
 
         if (languageConfig?.railroad) {
-            const diagram = createGrammarDiagramHtml(grammar);
             const diagramPath = path.resolve(relPath, languageConfig.railroad.out);
-            log('log', options, `Writing railroad syntax diagram to ${chalk.white.bold(diagramPath)}`);
-            await writeWithFail(diagramPath, diagram, options);
+            // Single File or no info -> write to HTML.
+            if (languageConfig.railroad.mode !== 'svg') {
+                const diagram = createGrammarDiagramHtml(grammar);
+                log('log', options, `Writing railroad syntax diagram to ${chalk.white.bold(diagramPath)}`);
+                await writeWithFail(diagramPath, diagram, options);
+            }
+            // Svg files requested -> make dir and write into it.
+            else {
+                const diagrams = createGrammarDiagramSvg(grammar);
+                log('log', options, `Writing railroad syntax diagrams to ${chalk.white.bold(diagramPath)}`);
+                for (const [name, diagram] of diagrams) {
+                    await writeWithFail(`${diagramPath}/${name}.svg`, diagram, options);
```

Comment:
Suggestion: Please use `path.join` or `path.resolve` to join file system paths. This kind of joining will lead to problems in case the given directory ends in `/`.

Reply:
Now using `path.join`.